File: 卡码/代码随想录/Python/二叉树/01-二叉树遍历方式/LeetCode_0116.py
#116. 填充每个节点的下一个右侧节点指针
from typing import *

class Node:
  def __init__(self, val: int = 0, left: 'Node' = None, right: 'Node' = None, next: 'Node' = None):
    self.val = val
    self.left = left
    self.right = right
    self.next = next

class Solution:
  def connect(self, root: 'Optional[Node]') -> 'Optional[Node]':
    q:List[Node]=[]
    cur:Node=None
    pre:Node=None
    if root!=None:
      q.append(root)
    while len(q)>0:
      size:int=len(q)
      for i in range(size):
        cur=q[0]
        q=q[1:]
        if i!=0:
          pre.next=cur
        pre=cur
        if cur.left!=None:
          q.append(cur.left)
        if cur.right!=None:
          q.append(cur.right)
    return root

File: 卡码/代码随想录/Go/二叉树/01-二叉树遍历方式/LeetCode_0116.go
//116. 填充每个节点的下一个右侧节点指针
package main

type Node struct {
	Val int
	Left *Node
  Right *Node
  Next *Node
}

func connect(root *Node) *Node {
	q:=make([]*Node,0)
	var cur,pre *Node
	if root!=nil {
		q=append(q,root)
	}
	for len(q)>0 {
		size:=len(q)
		for i:=0;i<size;i++ {
			if i==0 {
				pre=q[0]
				q=q[1:]
				cur=pre
			}else{
				cur=q[0]
				q=q[1:]
				pre.Next=cur
				pre=cur
			}
			if cur.Left!=nil {
				q=append(q,cur.Left)
			}
			if cur.Right!=nil {
				q=append(q,cur.Right)
			}
		}
		pre.Next=nil
	}
	return root
}
